mmv1/third_party/terraform/services/dataproc/resource_dataproc_session_template_test.go

```diff
@@ -0,0 +1,198 @@
+package dataproc_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccDataprocSessionTemplate_dataprocSessionTemplatesJupyterFullExample_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"project_name":    envvar.GetTestProjectFromEnv(),
+		"kms_key_name":    acctest.BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", "us-central1", "tf-bootstrap-dataproc-session-template-key1").CryptoKey.Name,
+		"prevent_destroy": false,
+		"subnetwork_name": acctest.BootstrapSubnetWithFirewallForDataprocBatches(t, "jupyer-session-test-network", "jupyter-session-test-subnetwork"),
+		"random_suffix":   acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckDataprocSessionTemplateDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataprocSessionTemplate_dataprocSessionTemplatesJupyterFullExample(context),
+			},
+			{
+				ResourceName:            "google_dataproc_session_template.dataproc_session_templates_jupyter_full",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "location", "runtime_config.0.properties", "terraform_labels"},
+			},
+			{
+				Config: testAccDataprocSessionTemplate_dataprocSessionTemplatesJupyterFullExample_update(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_dataproc_session_template.dataproc_session_templates_jupyter_full", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_dataproc_session_template.dataproc_session_templates_jupyter_full",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "location", "runtime_config.0.properties", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccDataprocSessionTemplate_dataprocSessionTemplatesJupyterFullExample_full(context map[string]interface{}) string {
```

Comment:
This seems less "full" compared to the generated full example. Is there a better name for this? Maybe `testAccDataprocSessionTemplate_preupdate` and then `testAccDataprocSessionTemplate_updated`? just for the purposes of following this test